tools/building.py

```diff
@@ -283,12 +283,9 @@ def lld_flags_for_executable(external_symbols):
   if settings.SHARED_MEMORY:
     cmd.append('--shared-memory')
 
-  # wasm-ld can strip debug info for us. this strips both the Names
-  # section and DWARF, so we can only use it when we don't need any of
-  # those things.
-  if settings.DEBUG_LEVEL < 2 and (not settings.EMIT_SYMBOL_MAP and
-                                   not settings.EMIT_NAME_SECTION and
-                                   not settings.ASYNCIFY):
+  # wasm-ld can strip debug info for us. This strips only DWARF, and not
+  # the name section.
```

Comment:
if name section stripping is necessary, why don't we just use `--strip-all` here then?

Reply:
I think that might be related to the more complex cases where we need debug info in intermediate files (like for Asyncify which reads the name section). Only `emcc.py` knows if we are at the point where it is ok to remove the names section, not `building.py`, so `building.py` leaves that to `emcc.py`.